ImputationPipeline/AggregatePRSResults.wdl

```diff
@@ -0,0 +1,235 @@
+    File batch_summarised_results = glob("*_summarised_results.tsv")[0]
+    File batch_score_distribution = glob("*_score_distribution.png")[0]
+  }
+}
+
+task PlotPCA {
+  input {
+    Array[File] target_pc_projections
+    File population_pc_projections
+    String batch_id
+    String population_name
+  }
+
+  command <<<
+    Rscript - <<- "EOF"
+    library(dplyr)
+    library(readr)
+    library(purrr)
+    library(ggplot2)
+
+    target_pcs <- c("~{sep='","' target_pc_projections}") %>% map(read_tsv) %>% reduce(bind_rows)
+    population_pcs <- read_tsv("~{population_pc_projections}")
+
+    ggplot(population_pcs, aes(x=PC1, y=PC2, color="~{population_name}")) +
+      geom_point(size=0.1, alpha=0.1) +
+      geom_point(data=target_pcs, aes(color="~{batch_id}")) +
+      theme_bw()
+
+    ggsave(filename = "~{batch_id}_PCA_plot.png", dpi=300, width = 6, height = 6)
+
+    EOF
+
+  >>>
+
+  runtime {
+    docker: "rocker/tidyverse@sha256:aaace6c41a258e13da76881f0b282932377680618fcd5d121583f9455305e727"
+    disks: "local-disk 100 HDD"
+    memory: "4 GB"
+  }
+
+  output {
+    File pc_plot = "~{batch_id}_PCA_plot.png"
+  }
+}
+
+task BuildHTMLReport {
+  input {
+    File batch_all_results
+    File batch_control_results
+    File expected_control_results
+    File batch_summarised_results
+    File score_distribution
+    File pc_plot
+    String batch_id
+  }
+
+  command <<<
+    set -xeo pipefail
+
+    cat << EOF > ~{batch_id}_report.Rmd
+    ---
+    title: "Batch ~{batch_id} PRS Summary"
+    output:
+    html_document:
+      df_print: paged
+    date: "$(date)"
+    ---
+
+    \`\`\`{r setup, include=FALSE}
+    library(readr)
+    library(ggplot2)
+    library(knitr)
+    library(dplyr)
+    library(stringr)
+    library(purrr)
+    batch_all_results <- read_tsv("~{batch_all_results}")
+    batch_control_results <- read_tsv("~{batch_control_results}")
+    expected_control_results <- read_csv("~{expected_control_results}")
+    batch_summary <- read_tsv("~{batch_summarised_results}")
+    batch_summary <- batch_summary %>% rename_with(.cols = -condition, ~ str_to_title(gsub("_"," ", .x)))
+    \`\`\`
+
+
+    ## Control Sample
+    \`\`\`{r control, echo = FALSE, results = "asis"}
+    kable(list(batch_control_results, expected_control_results) %>% reduce(bind_rows) %>% select(-batch_id) %>% select(-is_control) , digits = 2)
+    \`\`\`
+
+    ## Batch Summary
+    \`\`\`{r summary table, echo = FALSE, results = "asis" }
+    kable(batch_summary, digits = 2)
+    \`\`\`
+
+
+
+    ## Batch Score distribution
+    ![](~{score_distribution})
+
+    ## PCA
+    ![](~{pc_plot})
+
+    ## Individual Sample Results
+    \`\`\`{r sample results , echo = FALSE, results = "asis"}
+    kable(batch_all_results %>% mutate(across(ends_with("risk"), ~ kableExtra::cell_spec(.x, color=ifelse(is.na(.x), "blue", ifelse(.x=="NOT_RESULTED", "red", ifelse(.x == "HIGH", "orange", "green")))))), digits = 2)
```

Comment:
I think we should filter out the control sample from batch_all_results before displaying the table